lib/fizzy/execute.cpp

```diff
@@ -22,6 +22,63 @@ struct LabelContext
 };
 
 
+void matchImportedFunctions(const std::vector<TypeIdx>& module_imported_types,
+    const std::vector<ImportedFunction>& imported_functions)
+{
+    if (module_imported_types.size() != imported_functions.size())
+        throw std::runtime_error("Module requires " + std::to_string(module_imported_types.size()) +
+                                 " imported functions, " +
+                                 std::to_string(imported_functions.size()) + " provided");
+}
+
+void matchImportedGlobals(const std::vector<bool>& module_imports_mutability,
+    const std::vector<ImportedGlobal>& imported_globals)
+{
+    if (module_imports_mutability.size() != imported_globals.size())
+        throw std::runtime_error(
+            "Module requires " + std::to_string(module_imports_mutability.size()) +
+            " imported globals, " + std::to_string(imported_globals.size()) + " provided");
+
+    for (size_t i = 0; i < imported_globals.size(); ++i)
+    {
+        if (imported_globals[i].is_mutable != module_imports_mutability[i])
+            throw std::runtime_error("Global " + std::to_string(i) +
+                                     " mutability doesn't match module's global mutability");
+        if (imported_globals[i].value == nullptr)
+            throw std::runtime_error(
+                "Global " + std::to_string(i) + " has a null pointer to value");
+    }
+}
+
+std::vector<TypeIdx> matchImports(const Module& module,
+    const std::vector<ImportedFunction>& imported_functions,
+    const std::vector<ImportedGlobal>& imported_globals)
+{
+    std::vector<TypeIdx> imported_function_types;
+    std::vector<bool> imported_globals_mutability;
```

Comment:
Using of `vector<bool>` is sometimes discouraged because of its `operator[]` not returning a value in a contiguous segment of memory like for regular vector... But here it's a simple usage, I think it's acceptable.